instance/tests/models/test_instance.py

```diff
@@ -335,6 +340,29 @@ def test_run_playbook(self, mock_open_repo, mock_run_playbook, mock_inventory, m
             username='ubuntu',
         ), mock_run_playbook.mock_calls)
 
+    @patch('instance.models.instance.ansible.run_playbook')
+    def test_run_playbook_logging(self, mock_run_playbook):
+        """
+        Ensure logging routines are working on _run_playbook method
+        """
+        with tempfile.NamedTemporaryFile() as stdout:
+            with tempfile.NamedTemporaryFile() as stderr:
+
+                mock_run_playbook.return_value.__enter__.return_value.stdout = open(stdout.name, "rb")
+                mock_run_playbook.return_value.__enter__.return_value.stderr = open(stderr.name, "rb")
+                mock_run_playbook.return_value.__enter__.return_value.returncode = 0
+
+                stdout.write(b"HELLO\n")
+                stdout.flush()
+                stderr.write(b"HI\n")
+                stderr.flush()
+
+                instance = OpenEdXInstanceFactory()
+                log, returncode = instance._run_playbook("requirements", "playbook")
+
+                self.assertEqual(log, ['HELLO', 'HI'])
```

Comment:
There isn't any guarantee for the order of these two entries, even when flushing after writing to each line.  The OS is actually free to deliver them out of order, since the writes went to different streams.  You can simply use `self.assertItemsEqual()` instead of `self.assertEqual()` here.